src/org/rascalmpl/library/util/PathConfig.java

```diff
@@ -683,12 +701,16 @@ private static void addLibraryToSourcePath(URIResolverRegistry reg, IListWriter
             }
         }
 
-        if (!foundSrc) {
+        if (!foundSrc && !projectName.equals("rascal")) {
```

Comment:
#### Why is this extra condition needed?

`rascal` has three source roots:
  - `src/org/rascalmpl/library`
  - `test/org/rascalmpl/benchmark`
  - `test/org/rascalmpl/test/data`

The `test/**` roots aren't normally* packaged in the jar, though, so `foundSrc` becomes `false` in the previous loop, and the jar root is unnecessarily appended (which may look confusing to the user).

Before this PR, this code was never reached (in the case of `rascal`) due to an early return (see the removed code in the diff, lines 666-671 in the old file), so it wasn't an issue. After this PR, the early return is gone, so the extra condition is needed here to avoid the append.

(*) I've made changes to `pom.xml` in this PR to package all source roots, but that's just for testing and probably will be reverted.

Reply:
Maybe I'm confused, but they shouldn't be packaged in the jar (as that is what maven defines), but still show up in the search path of the interpreter?

Reply:
that's right. We will also be having more new source locations shortly:
* src/org/rascalmpl/tutor
* src/org/rascalmp/typepal
* src/org/rascalmpl/compiler

Reply:
> Maybe I'm confused, but they shouldn't be packaged in the jar (as that is what maven defines)

Correct, the `test/**` source roots aren't packaged in the jar.

> but still show up in the search path of the interpreter?

Per line 689, the source roots to be included in the path config aren't taken from the jar but from `RASCAL.MF`:

https://github.com/usethesource/rascal/blob/5e6b2ceebef09a1ef6550b7010ce68d31fca2dab/src/org/rascalmpl/library/util/PathConfig.java#L689